service/src/main/java/com/codeforcommunity/processor/authenticated/ProtectedSchoolProcessorImpl.java

```diff
@@ -19,23 +26,90 @@ public ProtectedSchoolProcessorImpl(DSLContext db) {
 
   @Override
   public SchoolListResponse getAllSchools(JWTData userData) {
-    List<School> schools =
-        db.select(SCHOOLS.ID, SCHOOLS.NAME, SCHOOLS.ADDRESS, SCHOOLS.COUNTRY)
+    List<SchoolSummary> schools =
+        db.select(SCHOOLS.ID, SCHOOLS.NAME, SCHOOLS.COUNTRY)
             .from(SCHOOLS)
             .where(SCHOOLS.HIDDEN.isFalse())
             .and(SCHOOLS.DELETED_AT.isNull())
-            .fetchInto(School.class);
+            .fetchInto(SchoolSummary.class);
 
     return new SchoolListResponse(schools, schools.size());
   }
 
   @Override
   public School getSchool(JWTData userData, long schoolId) {
-    return db.select(SCHOOLS.ID, SCHOOLS.NAME, SCHOOLS.ADDRESS, SCHOOLS.COUNTRY)
-        .from(SCHOOLS)
-        .where(SCHOOLS.HIDDEN.isFalse())
-        .and(SCHOOLS.DELETED_AT.isNull())
-        .and(SCHOOLS.ID.eq(schoolId))
-        .fetchOneInto(School.class);
+    School school =
+        db.select(SCHOOLS.ID, SCHOOLS.NAME, SCHOOLS.ADDRESS, SCHOOLS.COUNTRY, SCHOOLS.HIDDEN)
+            .from(SCHOOLS)
+            .where(SCHOOLS.DELETED_AT.isNull())
+            .and(SCHOOLS.ID.eq(schoolId))
+            .fetchOneInto(School.class);
+
+    List<SchoolContact> contacts = this.getSchoolContacts(schoolId);
+    school.setContacts(contacts);
+    return school;
+  }
+
+  @Override
+  public School createSchool(JWTData userdata, NewSchoolRequest newSchoolRequest) {
+    String name = newSchoolRequest.getName();
+    String address = newSchoolRequest.getAddress();
+    Country country = newSchoolRequest.getCountry();
+    Boolean hidden = newSchoolRequest.getHidden();
+
+    SchoolsRecord school =
+        db.selectFrom(SCHOOLS)
+            .where(SCHOOLS.NAME.eq(name))
```

Comment:
If I were to insert two schools into the DB with the same name, is that allowed at a DB level?